File: 20210816-0947/Archived_Code_20200906_1604/Backup/Code_17122019_Backup_Do_Not_Delete/java-revisited/src/GoF_Design_Patterns/Creational/Builder/BuilderClient.java
package GoF_Design_Patterns.Creational.Builder;

public class BuilderClient {

    public static void main(String[] args) {

        HouseBuilder builder = new ConcreteHouseBuilder();
        HouseBuilderDirector director = new HouseBuilderDirector(builder);
        System.out.println("director.construct() = " + director.construct());
        
        builder = new WoodenHouseBuilder();
        director = new HouseBuilderDirector(builder);
        System.out.println("director.construct() = " + director.construct());
    }
}

File: 20210816-0947/Archived_Code_20200906_1604/Backup/Code_26012020_Backup_Do_Not_Delete/GoF-DesignPatterns/Creational-Patterns/Builder/BuilderClient.cpp
#include "ConcreteHouseBuilder.h"
#include "WoodenHouseBuilder.h"
#include "HouseBuilder.h"
#include "HouseBuildDirector.h"
#include <iostream>

int main() {
    HouseBuilder *builder = new ConcreteHouseBuilder();
    HouseBuildDirector *houseBuildDirector = new HouseBuildDirector(builder);
    std::cout << houseBuildDirector->construct()->toString();
    return 0;
}
